DSG-2448: fix saving schema lists

jgnuvalence opened the pull request:
# Basics

## PR Type

Ticket: https://dol-ewf.atlassian.net/browse/DSG-2448
fix saving schema lists

- [ ] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

<!-- Please describe the current behavior and expected new behavior, or link to the relevant issue # -->

## Screenshot of the affected changes

the issue was an incorrect schema json generation in the frontend:
![image](https://github.com/Nuvalence/dsgov-web/assets/97063738/bc2dec80-f470-4b3a-9960-e3d38232a86b)

this is what it looks like after the change:
![image](https://github.com/Nuvalence/dsgov-web/assets/97063738/0ccbd67f-9e1f-4d7e-90ba-ad202998eb9c)

test in local:
![image](https://github.com/Nuvalence/dsgov-web/assets/97063738/982a1dc6-3a3b-49de-8b27-4e8bf2f33fde)
![image](https://github.com/Nuvalence/dsgov-web/assets/97063738/65c7ec91-4466-49b4-9050-f5ffac679278)


## Accessibility report

<!-- Accessibility is a high priority for this project, please include a screenshot of the accessibility report if applicable -->

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

thechechogomez reviewed (changes requested):
@jgnuvalence A couple of things:
1. When changing from `contentType: DynamicEntity` to a `contentType` that is not `DynamicEntity`, the property `entitySchema` persistes. 
<img width="531" alt="Screenshot 2023-12-04 at 10 13 28 AM" src="https://github.com/Nuvalence/dsgov-web/assets/60438413/59fd8825-d7a6-4d1d-8c2d-cb1e234afc98">

2. When opening the dialog to modify the component, the selected schema does not persist. 

https://github.com/Nuvalence/dsgov-web/assets/60438413/e931c8ec-9880-437b-a98c-c0fb7526dd5f

jgnuvalence commented:
> @jgnuvalence A couple of things:
> 
> 1. When changing from `contentType: DynamicEntity` to a `contentType` that is not `DynamicEntity`, the property `entitySchema` persistes.
> 
> <img alt="Screenshot 2023-12-04 at 10 13 28 AM" width="531" src="https://private-user-images.githubusercontent.com/60438413/287744467-59fd8825-d7a6-4d1d-8c2d-cb1e234afc98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3ODU0NDksIm5iZiI6MTcwMTc4NTE0OSwicGF0aCI6Ii82MDQzODQxMy8yODc3NDQ0NjctNTlmZDg4MjUtZDdhNi00ZDFkLThjMmQtY2IxZTIzNGFmYzk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDE0MDU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3ZWQyMjQ1ZWZmMWIwZmFhMjE0N2YwYjg1NTQyNmUyZjY0YmM5NTgwNWRmYWMyNThiYWQ2ZDJhMzEzNWVhYTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hI2fcwFkxQG2q90OQoAX_xJxKLYhkVCmXOutvFjeGw0">
> 2. When opening the dialog to modify the component, the selected schema does not persist.
> 
>  Screen.Recording.2023-12-04.at.10.18.05.AM.mov

thank you @thechechogomez I addressed the comments
